### PR TITLE
Add checked expr as oneof RBAC policy condition

### PR DESCRIPTION
**Pull request by jiangtaoli2016**
Commit Message: Add CheckedExpr as oneof expression in RBAC proto
Additional Description: CheckedExpr is type check CEL expression. It is future proof and offers better evaluation performance (not for C++ yet, but for other languages). There will be a separate PR to plumb checked expr to RBAC filter.
Risk Level: low
Testing: existing tests
Docs Changes:
Release Notes:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: api/envoy/config/rbac/v3/rbac.proto [hunk 1]
```diff
@@ -104,7 +105,13 @@ message Policy {
   // An optional symbolic expression specifying an access control
   // :ref:`condition <arch_overview_condition>`. The condition is combined
   // with the permissions and the principals as a clause with AND semantics.
-  google.api.expr.v1alpha1.Expr condition = 3;
+  oneof expression_specifier {
```
Comment (markdroth) [inline api/envoy/config/rbac/v3/rbac.proto:108]: Adding a `oneof` around an existing field is a breaking change in the golang proto API.  For now, you will need to add the new field without the `oneof`, but you can add an annotation indicating that in the next major version of the API, these two fields should be promoted into the same `oneof`.  I don't remember offhand how to do that, but @htuch can probably tell you.

Comment (jiangtaoli2016) [reply api/envoy/config/rbac/v3/rbac.proto]: changed to two separate fields with migrate tag.

**Comment by jiangtaoli2016** [comment]
@htuch I have trouble running the `tools/proto_format/proto_format.sh`, it gives the following errors
https://paste.googleplex.com/5656128161579008
Could you please help me figure out what is going on?

File: api/envoy/config/rbac/v3/rbac.proto [hunk 2]
```diff
@@ -104,7 +106,14 @@ message Policy {
   // An optional symbolic expression specifying an access control
   // :ref:`condition <arch_overview_condition>`. The condition is combined
   // with the permissions and the principals as a clause with AND semantics.
-  google.api.expr.v1alpha1.Expr condition = 3;
+  // Only be used when checked_condition is not used.
+  google.api.expr.v1alpha1.Expr condition = 3
+      [(udpa.annotations.field_migrate).oneof_promotion = "expression_specifier"];
+
+  // An optional symbolic expression that has been successfully type checked.
+  // Only be used when condition is not used.
```
Comment (htuch) [inline api/envoy/config/rbac/v3/rbac.proto:114]: Are you planning on plumbing this through to the RBAC filter in this PR?

Comment (jiangtaoli2016) [reply api/envoy/config/rbac/v3/rbac.proto:114]: I will make a follow-up PR to plumb into RBAC filter.

File: api/bazel/api_build_system.bzl [hunk 3]
```diff
@@ -188,5 +188,5 @@ def api_proto_package(
             "@com_envoyproxy_protoc_gen_validate//validate:go_default_library",
             "@com_google_googleapis//google/api:annotations_go_proto",
             "@com_google_googleapis//google/rpc:status_go_proto",
-        ],
+        ]).to_list(),
```
Comment (htuch) [inline api/bazel/api_build_system.bzl:191]: How does this change relate?

File: api/bazel/external_proto_deps.bzl [hunk 4]
```diff
@@ -17,20 +18,23 @@ EXTERNAL_PROTO_IMPORT_BAZEL_DEP_MAP = {
 
 # This maps from the Bazel proto_library target to the Go language binding target for external dependencies.
 EXTERNAL_PROTO_GO_BAZEL_DEP_MAP = {
+    "@com_google_googleapis//google/api/expr/v1alpha1:checked_proto": "@com_google_googleapis//google/api/expr/v1alpha1:expr_go_proto",
```
Comment (htuch) [inline api/bazel/external_proto_deps.bzl:21]: Ditto, why are we seeing reformats here?

File: api/envoy/config/rbac/v3/rbac.proto (hunk 2 above)
Comment (htuch) [reply api/envoy/config/rbac/v3/rbac.proto:114]: Can you add a `[#not-implemented-hide:]` comment above the field for now until that happens?

Comment (jiangtaoli2016) [reply api/envoy/config/rbac/v3/rbac.proto:115]: done

File: api/bazel/api_build_system.bzl (hunk 3 above)
Comment (jiangtaoli2016) [reply api/bazel/api_build_system.bzl:191]: In cel library, checked.proto and syntax.proto have two proto libraries, but it merges into one go_proto. It causes build error due to duplicated dependencies. I have to add depset().to_list() to remove duplicative entry.

File: api/bazel/external_proto_deps.bzl (hunk 4 above)
Comment (jiangtaoli2016) [reply api/bazel/external_proto_deps.bzl:21]: As explained above, proto has two libraries, but only has one go_proto.

**Comment by jiangtaoli2016** [comment]
@markdroth @htuch Thanks for feedbacks. This PR is ready for review.
@kyessenov FYI.

**Comment by jiangtaoli2016** [comment]
The failing tests seem unrelated to this PR.

File: api/bazel/api_build_system.bzl (hunk 3 above)
Comment (htuch) [reply api/bazel/api_build_system.bzl:195]: Can you add a comment above this to explain to future maintainers?

Comment (jiangtaoli2016) [reply api/bazel/api_build_system.bzl:198]: SG. I added a comment above.

**Comment by jiangtaoli2016** [comment]
@htuch Could you help merge?

**Pull request merged**